test/threads.jl

```diff
@@ -141,3 +141,40 @@ end
 @test_throws TypeError Atomic{Bool}
 @test_throws TypeError Atomic{BigInt}
 @test_throws TypeError Atomic{Float64}
+
+# Test atomic fences
+type CommBuf
+    data1::Atomic{Int}
+    data2::Atomic{Int}
+    correct::Bool
+    CommBuf() = new()
+end
+function test_fence_write(commbuf::CommBuf)
+    for i in 10000:-1:1
+        commbuf.data1[] = i
+        fence()
+        commbuf.data2[] = i
+        # data1 is always written before data2, so data1 <= data2
+    end
+end
+function test_fence_check(commbuf::CommBuf)
+    commbuf.correct = true
+    while true
+        data2 = commbuf.data2[]
+        data1 = commbuf.data1[]
+        commbuf.correct |= data1 <= data2
+        if data1 == 1 break end
+    end
+end
+function test_fence()
+    commbuf = CommBuf()
```

Comment:
Okay, at least `.correct` was initialized in a different routine.

I re-wrote the test, and everything should be properly initialized now.